pkg/probe/probes.go

```diff
@@ -96,54 +118,137 @@ func checkNodeReadiness(client client.Client, timeout time.Duration) error {
+			log.Info("default gw missing", "state", gjsonCurrentState.String())
 			return false, nil
 		}
 
 		return true, nil
 	})
 }
 
-func RunAll(client client.Client) error {
+func runPing(client client.Client, timeout time.Duration) error {
 	defaultGw, err := defaultGw()
 	if err != nil {
 		return errors.Wrap(err, "failed to retrieve default gw at runProbes")
 	}
 
-	currentState, err := nmstatectl.Show()
+	// TODO: Make ping timeout configurable with a config map
+	pingOutput, err := ping(defaultGw, timeout)
 	if err != nil {
-		return errors.Wrap(err, "failed to retrieve currentState at runProbes")
+		return errors.Wrapf(err, "error pinging default gateway -> output: %s", pingOutput)
 	}
+	return nil
+}
 
-	// TODO: Make ping timeout configurable with a config map
-	pingOutput, err := ping(defaultGw, defaultGwProbeTimeout)
+func runDNS(client client.Client, timeout time.Duration) error {
+	currentStateAsGJson, err := currentStateAsGJson()
 	if err != nil {
-		return errors.Wrapf(err, "error pinging external address after network reconfiguration -> output: %s, currentState: %s", pingOutput, currentState)
+		return errors.Wrap(err, "failed retrieving current state to get name resolving config")
 	}
 
-	err = checkApiServerConnectivity(apiServerProbeTimeout)
-	if err != nil {
-		return errors.Wrapf(err, "error checking api server connectivity after network reconfiguration -> currentState: %s", currentState)
+	// Get the name servers at node since the ones at container are not accurate
+	runningServersGJsonPath := "dns-resolver.running.server"
+	runningNameServers := currentStateAsGJson.Get(runningServersGJsonPath).Array()
+	if len(runningNameServers) == 0 {
+		return fmt.Errorf("missing name servers at '%s' on %s", runningServersGJsonPath, currentStateAsGJson.String())
 	}
 
-	err = checkNodeReadiness(client, nodeReadinessProbeTimeout)
+	// Check that at least one of the name servers can resolve root server NS
+	rootNS := "root-server.net"
+	resolved := false
+	for _, runningNameServer := range runningNameServers {
+		r := &net.Resolver{
+			PreferGo: true,
+			Dial: func(ctx context.Context, network, address string) (net.Conn, error) {
+				d := net.Dialer{
+					Timeout: timeout,
+				}
+				return d.DialContext(ctx, network, net.JoinHostPort(runningNameServer.String(), "53"))
+			},
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		_, err := r.LookupNS(ctx, rootNS)
+		if err != nil {
+			log.Info("WARNING cannot resolve root server NS", "rootNS", rootNS, "nameServer", runningNameServer.String(), "err", err)
+		} else {
+			resolved = true
+			break
+		}
+	}
+	if !resolved {
+		return fmt.Errorf("failed to query NS of %s with name servers %+v", rootNS, runningNameServers)
+	}
+	return nil
+}
+
+// Select will return the external connectivity probes that are working (ping and dns) and
+// the internal connectivity probes
+func Select(cli client.Client) []Probe {
```

Comment:
SelectProbe maybe ?

Reply:
Probe is already part of the package name so we use it like `probe.Select` so no need to put Probe there again.